src/journal/ObjectRecorder.cc

```diff
@@ -263,6 +267,7 @@ void ObjectRecorder::handle_append_flushed(uint64_t tid, int r) {
 
       // notify of overflow once all in-flight ops are complete
       if (m_in_flight_tids.empty() && !m_aio_scheduled) {
+        m_append_buffers.splice(m_append_buffers.begin(), m_pending_buffers);
```

Comment:
Nit: could move this logic to ```notify_handler_unlock``` so it's only in the code once.

Reply:
@dillaman I think it should be before `append_overflowed()`, otherwise the appends may be in wrong order (inflight appends are inserted at the begin of m_append_buffers in `append_overflowed()`, so at that stage `m_append_buffers` should already contain `m_pending_buffers`)?

I think can move this to `append_overflowed()`, but then it is still needed in the block below.

Reply:
Yeah, you're correct. No worries about changing this then.